agent/unit-test/configstore_unittest.cpp

```diff
@@ -0,0 +1,119 @@
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#include <configstore.h>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <unistd.h>
+
+class ConfigStoreTest: public ::testing::Test
+{
+public:
+    LOG_DECLARE_CLASS_CONTEXT("TEST", "Tester");
+    ConfigStoreTest() { }
+    ConfigStore *cs = new ConfigStore("");
+    std::string capNameA = "com.pelagicore.temperatureservice.gettemperature";
+    std::string capNameB = "com.pelagicore.temperatureservice.settemperature";
+    std::string capNameC = "dummyCapC";
+    std::string capNameD = "dummyCapD";
+
+    /* The Service Manifest (absolute) file paths */
+    std::string homeDir       = "/home/vagrant/softwarecontainer/agent/unit-test/";
+    std::string sm_path       = "CS_unittest_ServiceManifest.json";
+    std::string short_sm_path = "CS_unittest_short_ServiceManifest.json";
+    std::string evil_sm       = "CS_unittest_parseError.json";
+    std::string inputDir      = "/home/vagrant/";
+
+    /* Setting up the environment */
+    void SetUp() override
+    {
+        // Code here will be called right before each test
+    }
+
+    /* Destroy objects created */
+    void TearDown() override
+    {
+        // Code here will be called right before the destructor.
+    }
+};
+
+/* The tests */
+TEST_F(ConfigStoreTest, constructorNullstr) {
+    ASSERT_NO_THROW(ConfigStore(""));
+}
+
+TEST_F(ConfigStoreTest, constructorFileOk) {
+    ASSERT_NO_THROW(ConfigStore(homeDir+short_sm_path));
+}
+
+TEST_F(ConfigStoreTest, constructorFileOk2) {
+    ASSERT_NO_THROW(ConfigStore(homeDir+sm_path));
+}
+
+TEST_F(ConfigStoreTest, constructorEvilFile) {
+    ASSERT_THROW(ConfigStore(homeDir+evil_sm), ReturnCode);
+}
+
+TEST_F(ConfigStoreTest, constructorDir) {
+    ASSERT_NO_THROW(ConfigStore(homeDir+""));
+}
+
+/* This test doesn't work, it exits without obvious reason
+TEST_F(ConfigStoreTest, constructorDir2) {
+    ASSERT_NO_THROW(ConfigStore("/home/vagrant/"));
+}
+*/
+
+TEST_F(ConfigStoreTest, constructorEvilDir) {
+    // No such dir
+    ASSERT_ANY_THROW(ConfigStore("/home/tester"));
+}
+
+TEST_F(ConfigStoreTest, constructorEvilDir2) {
+    // '/' is not accepted as a dir for Service Manifests
+    ASSERT_ANY_THROW(ConfigStore("/"));
+}
+
+TEST_F(ConfigStoreTest, readCapsFromile) {
+    ASSERT_FALSE(isError(cs->readCapsFromFile(homeDir+short_sm_path)));
+}
+
+TEST_F(ConfigStoreTest, readConfigFetchOneCap) {
+    EXPECT_FALSE(isError(cs->readCapsFromFile(homeDir+short_sm_path)));
+
+    GatewayConfiguration retGWs = cs->readConfigs(capNameA);
+    ASSERT_FALSE(retGWs.empty());
+}
+
+TEST_F(ConfigStoreTest, readConfigFetchEvilCap) {
+    ASSERT_FALSE(isError(cs->readCapsFromFile(homeDir+sm_path)));
+
+    GatewayConfiguration retGWs = cs->readConfigs("EvilCapName");
+    ASSERT_TRUE(retGWs.empty());
+}
+
+TEST_F(ConfigStoreTest, readEvilConfig) {
+    ASSERT_TRUE(isError(cs->readCapsFromFile(homeDir+evil_sm)));
+}
```

Comment:
In general: add comments of what is tested, i.e. why would some calls be false etc. Also add spaces around operators like '+'